stress_test/controller.py

```diff
@@ -0,0 +1,402 @@
+
+        super(self.__class__, self).__init__(ctrl_base_dir, test_config)
+
+        self.stat_period_ms = test_config['controller_statistics_period_ms']
+
+        if 'controller_flowmods_conf_handler' in test_config:
+            self.flowmods_conf_hnd= ctrl_base_dir + test_config['controller_flowmods_conf_handler']
+
+        if 'controller_statistics_handler' in test_config:
+            self.statistics_hnd = ctrl_base_dir + test_config['controller_statistics_handler']
+
+        if 'controller_persistent_handler' in test_config:
+            self.persistent_hnd = ctrl_base_dir + test_config['controller_persistent_handler']
+
+        if 'controller_restconf_port' in test_config:
+            self.restconf_port = test_config['controller_restconf_port']
+            self.restconf_user = test_config['controller_restconf_user']
+            self.restconf_password = test_config['controller_restconf_password']
+
+        self.hosts = -1
+        self.switches = -1
+        self.links = -1
+        self.flows = -1
+
+
+    def generate_xmls(self, controller_cpus):
+        """ Starts and then stops the controller to trigger the generation of
+        controller's XML files.
+
+        :param controller_cpus: number of cpus returned by create_cpu_shares() and
+        allocated for controller
+        :type controller_cpus: str
+        """
+        logging.info('[Controller] Generating XMLs')
+        pid = self.start(controller_cpus)
+
+        # Controller status check is done within start()
+        logging.info('Controller status is 1.')
+        logging.info('[generate_xmls] Stopping controller')
+        self.stop()
+
+
+
+    def disable_persistence(self):
+        """configure controller persistent to false in order not to backup
+        datastore on the disk.
+        """
+        logging.info('[controller] Disabling persistent')
+
+        common.command_exec_wrapper([self.persistent_hnd],
+                                    '[controller_change_persistent_handler]',
+                                    self._init_ssh)
+
+
+    def change_stats(self):
+        """Wrapper to the controller statistics handler
+        """
+
+        logging.info('[controller] Changing statistics period')
+        common.command_exec_wrapper(
+            [self.statistics_hnd, str(stat_period_ms)],
+            '[controller.statistics_handler] Changing statistics interval',
+            self._init_ssh)
+        logging.info(
+            '[change_stats] Changed statistics period to {0} ms'.
+            format(self.stat_period_ms))
+
+
+    def flowmods_config(self):
+        """configure controller to send flow modifications as a responce to ARP
+        Packet_INs.
+        """
+        logging.info('[controller] Configure flow modifications')
+        common.command_exec_wrapper([self.flowmods_conf_hnd],
+                                    '[controller_flowmod_configure_handler]',
+                                    self._init_ssh)
+
+
```

Comment:
Note for `get_oper_{hosts, switches, links, flows}` methods: 

Leave the implementation of those methods currently blank, as these methods should call handlers, similar to start, stop etc, to get the hosts, switches, etc counts. 

See user story https://www.pivotaltracker.com/n/projects/1443942/stories/128045723